plugins/policy/cache/match_expression.go

```diff
@@ -97,17 +97,14 @@ func (pc *PolicyCache) getMatchExpressionPods(namespace string, expressions []*p
 	notExistPodSet = utils.RemoveDuplicates(inPodSet)
 
 	inMatcher := utils.Intersect(inPodSet, notInPodSet)
-	if inMatcher == nil {
-		return false, nil
+	if len(inMatcher) == 0 {
+		return []string{}
 	}
 	existsMatcher := utils.Intersect(existsPodSet, notExistPodSet)
-	if existsMatcher == nil {
-		return false, nil
+	if len(existsMatcher) == 0 {
+		return []string{}
 	}
 
 	pods := utils.Intersect(inMatcher, existsMatcher)
-	if pods == nil {
-		return false, nil
-	}
-	return true, pods
+	return pods
```

Comment:
same here, I would say update `utils.Intersect` to make it smarter on either of them is empty case, like 
```
func Intersect(a []string, b []string) []string {
   if len(a) == 0 || len(b) == 0 {
          return []string{}
   }
    ...
}
```
or further improve it to take a slice variables
```
// Intersect at least need two lists
func Intersect(a []string, b []string, s ...[]string) []string {
	if len(a) == 0 || len(b) == 0 {
		return []string{}
	}
	set := make([]string, 0)
	hash := make(map[string]bool)
	for _, el := range a {
		hash[el] = true
	}
	for _, el := range b {
		if _, found := hash[el]; found {
			set = append(set, el)
		}
	}
	if len(s) == 0 {
		return set
	}
	return Intersect(set, s[0], s[1:]...)
}
```
In this way, *a lot* places can be just
```
return utils.Intersect(inPodSet, notInPodSet, existsPodSet, notExistPodSet)
```